suite-native/fiat-rates/src/fiatRatesSelectors.ts

```diff
@@ -0,0 +1,207 @@
+    return currentRate?.isLoading ?? false;
+};
+
+export const selectIsTickerLoading = (
+    state: FiatRatesRootState,
+    ticker: TickerId,
+    fiatCurrency: FiatCurrencyCode,
+) => {
+    const fiatRateKey = getFiatRateKeyFromTicker(ticker, fiatCurrency);
+    return selectIsFiatRateLoading(state, fiatRateKey);
+};
+
+export const selectShouldUpdateFiatRate = (
+    state: FiatRatesRootState,
+    fiatRateKey: FiatRateKey,
+    rateType: RateType = 'current',
+) => {
+    const currentRate = selectFiatRatesByFiatRateKey(state, fiatRateKey, rateType);
+
+    if (!currentRate) {
+        return true;
+    }
+
+    const { lastSuccessfulFetchTimestamp } = currentRate;
+    const now = Date.now();
+
+    return now - lastSuccessfulFetchTimestamp > MAX_AGE[rateType];
+};
+
+export const selectTickerFromAccounts = (state: FiatRatesRootState): TickerId[] => {
+    const accounts = selectAccounts(state as any);
+    return pipe(
+        accounts,
+        A.map(account => [
+            {
+                symbol: account.symbol,
+            } as TickerId,
+            ...(account.tokens || []).map(
+                token =>
+                    ({
+                        symbol: token.symbol,
+                        mainNetworkSymbol: account.symbol,
+                        tokenAddress: token.address,
+                    } as TickerId),
+            ),
+        ]),
+        A.flat,
+        F.toMutable,
+    );
+};
+
+export const selectShouldUpdateTicker = (
+    state: FiatRatesRootState,
+    ticker: TickerId,
+    fiatCurrency: FiatCurrencyCode,
+) => {
+    const fiatRateKey = getFiatRateKeyFromTicker(ticker, fiatCurrency);
+    return selectShouldUpdateFiatRate(state, fiatRateKey);
+};
+
+export const selectTickersToBeUpdated = (
+    state: FiatRatesRootState,
+    fiatCurrency: FiatCurrencyCode,
+    rateType: RateType,
+): TickerId[] => {
+    const tickers = selectTickerFromAccounts(state);
+    return tickers.filter(ticker => {
+        const fiatRateKey = getFiatRateKeyFromTicker(ticker, fiatCurrency);
+        return (
+            selectShouldUpdateFiatRate(state, fiatRateKey, rateType) &&
+            !selectIsTickerLoading(state, ticker, fiatCurrency)
+        );
+    });
+};
+
+export const selectTransactionsWithMissingRates = memoizeWithArgs(
+    (state: FiatRatesRootState, localCurrency: FiatCurrencyCode) => {
+        const accountTransactions = selectTransactions(state);
+
+        return pipe(
+            accountTransactions,
+            D.mapWithKey((accountKey, txs) => ({
+                account: selectAccountByKey(state, accountKey as AccountKey),
+                txs: txs.filter(tx => !tx.rates?.[localCurrency]),
+            })),
+            D.filter(({ account, txs }) => !account || !txs.length),
+            D.values,
+            A.filter(value => !!value),
+        ) as {
+            account: Account;
+            txs: WalletAccountTransaction[];
+        }[];
+    },
+    { size: 1 },
```

Comment:
I want to keep it, because it's now clearly visible that there is something like that that you need keep in mind even when using selector for example.